### PR TITLE
remove deprecated shorthand flag

### PR DESCRIPTION
**Pull request by ChiuWang**
**Release note**:
```release-note
remove deprecated shorthand flag `-c` from `kubectl version (--client)`
```

### CODE REVIEW

File: pkg/kubectl/cmd/version.go
Comment (neolit123) [inline pkg/kubectl/cmd/version.go]: use `BoolVar()`.

**Comment by neolit123** [comment]
seems like a valid change, given the deprecation happened back in 1.9.0:
https://github.com/charrywanganthony/kubernetes/commit/6290aee8b8e1d7923fda953cf36200c5d06a42ac

TBH the `"short"` cleanup should have been outside of this PR, but i would leave it to the maintainers to decide.
/lgtm

**Comment by charrywanganthony** [comment]
cc @soltysh for approve.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
